overlord/devicestate/handlers_install.go

```diff
@@ -1413,6 +1424,16 @@ func (m *DeviceManager) doInstallSetupStorageEncryption(t *state.Task, _ *tomb.T
 		return err
 	}
 
+	// HACK: an unset revision means the snap will be local
+	// TODO: also do this for gadget
+	// TODO2: this also means we will leave pc-kernel snaps around
+	//        like /var/lib/snapd/snaps/pc-kernel_5.1.1.snap
+	//        because they will be unaccounted for(?)
+	kernelInfo := snapInfos[snap.TypeKernel]
+	if kernelInfo.Revision.Unset() {
```

Comment:
We actually do something very similar in the seedwriter code so I suspect this is not actually a hack, maybe it needs a different place though. I fixed TODO2 in my api-installer-finish-mvo branch in https://github.com/alfonsosanchezbeato/snapd/pull/4/commits/ef77a049dd943629a3898c1211ba16a8ab213793 (just FYI :)